compact.go

```diff
@@ -238,6 +247,33 @@ func (c *LeveledCompactor) selectDirs(ds []dirMeta) []dirMeta {
 	return nil
 }
 
+// selectOverlappingDirs returns one set of dir metas of the blocks whose ranges are overlapping,
+// that should be compacted into a single new block.
+func (c *LeveledCompactor) selectOverlappingDirs(ds []dirMeta) []dirMeta {
```

Comment:
I think this is really connected with https://github.com/prometheus/prometheus/pull/4653

Basically this creates irregular blocks right?

Reply:
This returns the `dirMeta` of the blocks which are overlapping. This is used as the first check of compactor `Plan`. (Basically giving higher priority in selecting overlapping blocks before any other compaction)